experiment/src/org/labkey/experiment/xar/FolderXarWriterFactory.java

```diff
@@ -98,29 +98,36 @@ public boolean show(Container c)
             return children;
         }
 
-        private List<ExpRun> getRuns(@Nullable FolderExportContext ctx, Container c)
+        private boolean hasRuns(Container c)
+        {
+            return ExperimentService.get().hasExpRuns(c, this::runFilter);
+        }
+
+        private boolean runFilter(ExpRun run) {
+            return !run.getProtocol().getLSID().equals(ExperimentService.SAMPLE_DERIVATION_PROTOCOL_LSID)
+                    && !run.getProtocol().getLSID().equals(ExperimentService.SAMPLE_ALIQUOT_PROTOCOL_LSID)
+                    && !"recipe".equalsIgnoreCase(run.getProtocol().getImplementationName())
+                    && !"recipe".equalsIgnoreCase(run.getProtocol().getLSIDNamespacePrefix());
+        }
+
+        private List<ExpRun> getRuns(Container c)
         {
             // Don't include the sample derivation runs; we now have a separate exporter explicitly for sample types.
             // Also don't include recipe protocols; there's a separate folder writer and importer for the recipe module.
             // if an additional context has been furnished, filter out runs not included in this export
 
-            List<ExpRun> allRuns = (List<ExpRun>) ExperimentService.get().getExpRuns(c, null, null,
-                        run -> !run.getProtocol().getLSID().equals(ExperimentService.SAMPLE_DERIVATION_PROTOCOL_LSID)
-                                && !run.getProtocol().getLSID().equals(ExperimentService.SAMPLE_ALIQUOT_PROTOCOL_LSID)
-                                && !"recipe".equalsIgnoreCase(run.getProtocol().getImplementationName())
-                                && !"recipe".equalsIgnoreCase(run.getProtocol().getLSIDNamespacePrefix())
-                    );
+            List<ExpRun> allRuns = (List<ExpRun>) ExperimentService.get().getExpRuns(c, null, null, this::runFilter);

```

Comment:
Since this is inside of the Experiment module, you can use ExperimentServiceImpl to avoid needing to cast. You could also get away without needing to add `hasRuns()` to the interface if that's appealing

```suggestion
            List<ExpRun> allRuns = ExperimentServiceImpl.get().getExpRuns(c, null, null, this::runFilter);
```